File: Lecture-10/Lecture/Code/LinkedList/NodePart2.cpp
T Node::getValue() const {
  return _value;
}
void Node::setValue(T value) {
  _value = value;
}

Node<T>* Node::getNext() const {
  return _next;
}
void Node::setNext(Node<T>* next) {
  _next = next;
}

File: Lecture-10/Lecture/Code/LinkedList/NodePart2.java
	/**
	 * Creates a new node with a next node
	 * assigned.
	 */
	public Node(T @{\color{java_variable}value}@, Node<T> @{\color{java_variable}next}@) {
		this.value = @{\color{java_variable}value}@;
		this.next = @{\color{java_variable}next}@;
	}
	/**
	 * Creates a new node without a next
	 * node assigned.
	 */
	public Node(T @{\color{java_variable}value}@) {
		this.value = @{\color{java_variable}value}@;
		this.next = null;
	}